lib/datadog/statsd/udp_connection.rb

```diff
@@ -37,7 +36,12 @@ def connect
         @socket.connect(host, port)
       end
 
+      # send_message is writing the message in the socket, it may create the socket if nil
+      # It is not thread-safe but since it is called by either the Sender bg thread or the
+      # SingleThreadSender (which is using a mutex while Flushing), only one thread should call
+      # it at a time.
```

Comment:
Minor: I hadn't seen this when I commented on the comment added to `Connection` but this comment is indeed closer to what I suggested there as well :)

I would maybe not use _should_ in "only one thread should call", but would instead use `_must_` (see also https://datatracker.ietf.org/doc/html/rfc2119).